Userland/Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -1877,6 +1877,9 @@ void perform_url_and_history_update_steps(DOM::Document& document, AK::URL new_u
         // 1. Finalize a same-document navigation given traversable, navigable, newEntry, and entryToReplace.
         finalize_a_same_document_navigation(*traversable, *navigable, new_entry, entry_to_replace);
     });
+
+    // FIXME: add synchronous history update steps as in this WIP commit. (as in https://github.com/SerenityOS/serenity/commit/a75650993df2532f5c3f05856fdde28b062b0ae1)
```

Comment:
let's avoid links to commits from not merged branches. otherwise the change looks good!

Reply:
sure! changed comment text, thank you!